### ISSUE TITLE
packbits pads least significant bits, rather than most significant

### ISSUE THREAD

**Issue by 1rocketdude** [numpy/numpy#9398]
windows 7, python 3.5.3, numpy 1.13.0

I would expect that packbits were pad zeros to the most significant bits in a word that is less than 8-bits (or multiples of 8); e.g.,
```
c=np.array((1,0,0,0,0),dtype=np.uint8)
np.packbits(c)
```

should return 32, but returns 128 because the 6 bits 'are padded with 2 zeros in index 6 & 7 before computing the value of 128. If padding is necessary, I would think the more appropriate thing to do is pad zeros at the most significant bits 0 and 1.

unpackbits appears to have the same behavior (padding bits 6 & 7 with zeros)
```
d=np.array((32),dtype=np.uint8)
np.unpackbits(d)
```
array([0, 0, 1, 0, 0, 0, 0, 0], dtype=uint8)

**Comment by pv** [comment]
This would be a semantic change in a function that existed for 10 years, 
and the reason given looks cosmetic (lsb vs msb). Looks like wontfix.

**Comment by eric-wieser** [comment]
Seems to me that `np.packbits([1, 0, 0 ,0, 0])` should actually be `1`, in the same way that `Polynomial([1, 0, 0])` is `1`.

So maybe we should introduce a fixed version of `packbits` in the same way that `np.polynomial.Polynomial` is a fixed version of `np.poly1d`?

**Comment by juliantaylor** [comment]
Yeah this is a little odd, due to being msb it is also slower on common architectures as the equivalent machine instructions on little endian machines are based ib lsb.
A keyword argument to switch it might be an option, though having an specific usecase for it would be interesting first.

**Comment by juliantaylor** [comment]
The masks of CASA tables are lsb packed bits, but you wouldn't want to read those directly with numpy. But maybe there are other formats that could use the flexibility? Then you could just byteswap before unpacking/packing so the gain is not really that great.

**Comment by eric-wieser** [comment]
If we did a v2 of these functions, then supporting `uint32 -> bits` would be nice, rather than going through `reshape` and `view((uint8, 4))`. Also, returning bool rather than `uint8`

**Comment by jpivarski** [comment]
Here's a use-case: [Apache Arrow null bitmaps](https://github.com/apache/arrow/blob/master/format/Layout.md#null-bitmaps) are encoded LSB. I'm casting them with `frombuffer` and interpreting them with Numpy.

The conversion is easy to implement but hard to guess ([thanks, StackOverflow!](https://stackoverflow.com/a/46058378/1623645)):

```
np.unpackbits(fbyte).reshape(-1,8)[:,::-1]
```

I'd be in favor of a parameter that toggles between MSB (the default for many years) and LSB. At the very least, the existence of a toggle would alert the user that there's an issue to be aware of.

**Comment by zhuanhao-wu** [comment]
I also wonder if we could add this feature.

I'm trying to generate 64-bit integer and want to extract 4-bit group out of it and feed the 4-bit data to an FPGA or a chip.

The data bus might be 32-bit and packing bits in LSB would be really helpful.

**Comment by SimonSegerblomRex** [comment]
This issue can be closed referencing #12962.
